server/controllers/parties.js

```diff
@@ -1,97 +1,118 @@
+    const { name, hqaddress, logourl } = req.body;
+    const values = [name, hqaddress, logourl];
+    db.query('SELECT * FROM parties WHERE name = $1', [name], (error, response) => {
+      if (response.rowCount !== 0) {
+        return res.status(400).json({
+          status: 400,
+          error: 'party already exists',
+        });
+      }
+      return db.query(text, values, (err, newParty) => {
+        if (err) {
+          res.status(400).send({
+            status: 400,
+            error: err.stack,
+          });
+        }
+        res.status(201).send({
+          status: 201,
+          data: [newParty.rows[0]],
+        });
       });
-    }
-    const createdParty = {
-      id: partiesDb.length + 1,
-      name: req.body.name,
-      hqAddress: req.body.hqAddress,
-      logoUrl: req.body.logoUrl,
-    };
-    partiesDb.push(createdParty);
-    return res.status(201).json({
-      status: 201,
-      data: [
-        {
-          success: true,
-          message: 'Party created successfully',
-          data: createdParty,
-        },
-      ],
     });
   }
 
   // READ - Get All Parties
   static getAll(req, res) {
-    return res.status(200).json({
-      status: 200,
-      data: partiesDb,
+    db.query('SELECT * FROM parties ORDER BY id ASC', (err, foundParties) => {
+      // callback
+      if (err) {
+        res.status(400).json({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      return res.status(200).send({
+        status: 200,
+        data: foundParties.rows,
+      });
     });
   }
 
   // READ - Get a particular party
   static getOne(req, res) {
+    const text = 'SELECT * FROM parties WHERE id = $1';
     const id = parseInt(req.params.id, 10);
-    const selectedParty = partiesDb.find(party => party.id === id);
-    if (!selectedParty) {
-      return res.status(404).json({
+    db.query(text, [id], (err, foundParty) => {
+      if (err) {
+        res.status(400).json({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      if (foundParty.rowCount !== 0) {
+        return res.status(200).json({
+          status: 200,
+          data: [foundParty.rows[0]],
+        });
+      }
+      return res.status(404).send({
         status: '404',
-        error: 'The party was not found',
+        error: 'The office was not found',
       });
-    }
-    return res.status(200).json({
-      status: 200,
-      data: [selectedParty],
     });
   }
 
   // PUT - Update Party
   static update(req, res) {
     const id = parseInt(req.params.id, 10);
-    const partyIndex = partiesDb.findIndex(party => party.id === id);
-
-    if (partyIndex === -1) {
-      return res.status(404).json({
+    const { name } = req.body;
+    db.query('UPDATE parties SET name = $1 WHERE id = $2 RETURNING *', [name, id], (err, updatedParty) => {
+      if (err) {
+        res.status(400).send({
+          status: 400,
+          error: err.stack,
```

Comment:
Is the error message is `err.stack` descriptive enough as feedback for users? Would you rather return a custom error message.

Reply:
I'll change the error message

Reply:
I have implemented the feedback